app/models/miq_provision_virt_workflow.rb

```diff
@@ -225,15 +233,23 @@ def available_vlans_and_hosts(options = {})
       rails_logger('allowed_vlans', 0)
       # TODO: Use Active Record to preload this data?
       MiqPreloader.preload(hosts, :switches => :lans)
-      hosts.each do |h|
-        h.lans.each { |l| vlans[l.name] = l.name unless l.switch.shared? }
-      end
+      load_allowed_vlans(hosts, vlans)
       rails_logger('allowed_vlans', 1)
     end
 
     return vlans, hosts
   end
 
+  def load_allowed_vlans(hosts, vlans)
+    load_hosts_networks(hosts, vlans)
+  end
+
+  def load_hosts_networks(hosts, vlans)
```

Comment:
Can you leave this as `_vlans` instead of `_networks` for consistency?  I don't agree with us using `vlans` here (I like networks better personally) but the concept is called vlans everywhere else here.